src/test/java/com/box/sdk/BoxSignRequestTest.java

```diff
@@ -0,0 +1,256 @@
+    }
+
+    @Test
+    @Category(UnitTest.class)
+    public void getAllSignRequestsSucceeds() throws IOException {
+        final String fileId = "12345";
+        final String fileName = "Contract.pdf";
+        final String signerEmail = "example@gmail.com";
+        final String signRequestId = "12345";
+
+        final String prepareUrl = "https://prepareurl.com";
+
+        final String requestUrl = "/sign_requests";
+
+        String result = TestConfig.getFixture("BoxSignRequest/GetAllSignRequests200");
+
+        WIRE_MOCK_CLASS_RULE.stubFor(WireMock.get(WireMock.urlPathEqualTo(requestUrl))
+                .willReturn(WireMock.aResponse()
+                        .withHeader("Content-Type", "application/json")
+                        .withBody(result)));
+
+        Iterator<BoxSignRequest.Info> signRequests = BoxSignRequest.getAll(this.api).iterator();
+        BoxSignRequest.Info firstSignRequest = signRequests.next();
+
+        BoxFile.Info fileInfo = firstSignRequest.getSourceFiles().get(0);
+        BoxSignRequestSigner signer = firstSignRequest.getSigners().get(0);
+
+        Assert.assertEquals(prepareUrl, firstSignRequest.getPrepareUrl());
+        Assert.assertEquals(fileId, fileInfo.getID());
+        Assert.assertEquals(fileName, fileInfo.getName());
+        Assert.assertEquals(signerEmail, signer.getEmail());
+        Assert.assertEquals(signRequestId, firstSignRequest.getID());
+    }
+
+    @Test
+    @Category(UnitTest.class)
+    public void cancelSignRequestSucceeds() throws IOException {
+        final String signRequestId = "12345";
+
+        final String requestUrl = "/sign_requests/" + signRequestId + "/cancel";
+
+        String result = TestConfig.getFixture("BoxSignRequest/CancelSignRequest200");
+
+        WIRE_MOCK_CLASS_RULE.stubFor(WireMock.post(WireMock.urlPathEqualTo(requestUrl))
+                .willReturn(WireMock.aResponse()
+                        .withHeader("Content-Type", "application/json")
+                        .withBody(result)));
+
+        BoxSignRequest signRequest = new BoxSignRequest(this.api, signRequestId);
+        BoxSignRequest.Info signRequestInfo = signRequest.cancel();
+
+        Assert.assertEquals(BoxSignRequest.BoxSignRequestStatus.Cancelled, signRequestInfo.getStatus());
+    }
+
+    @Test
+    @Category(UnitTest.class)
+    public void resendSignRequestSucceeds() {
+        final String signRequestId = "12345";
+
+        final String requestUrl = "/sign_requests/" + signRequestId + "/resend";
+
+        WIRE_MOCK_CLASS_RULE.stubFor(WireMock.post(WireMock.urlPathEqualTo(requestUrl))
+                .willReturn(WireMock.aResponse()
+                        .withStatus(202)));
+
+        BoxSignRequest signRequest = new BoxSignRequest(this.api, signRequestId);
+
+        signRequest.resend();
+
+        WireMock.verify(1, postRequestedFor(urlPathEqualTo(requestUrl)));
+    }
+
+    @Test
+    @Category(IntegrationTestSignRequest.class)
```

Comment:
We should probably change this to `IntegrationTest.class` and fix all the integration tests later.